docs/content/docs/dev/testing.md

```diff
@@ -41,26 +41,33 @@ flutter test
 
 ### Infrastructure
 
-You need a fresh [`synapse` matrix backend](https://matrix-org.github.io/synapse/latest/) with the following settings included (in the `homeserver.yaml`):
+You need a fresh [`synapse` matrix backend](https://matrix-org.github.io/synapse/latest/).
 
-```yaml
+Please change `bind_addresses` of `listeners` from `['::1', '127.0.0.1']` to `['0.0.0.0']`, that means any address and allows remote connection (non-localhost).
+
+Please turn off the public profile of server firewall, that allows synapse to accept the request from external. If you are running `Ubuntu` linux, you can use `gufw` app to do it.
+
+![Ubuntu Firewall](../../../static/images/ubuntu-firewall.png)
```

Comment:
Let's move that entire section _below_ the `Docker Container` section into its own section of `#### Firewall`. It is confusing to find this here in between.

Reply:
Cool